### PR TITLE
:sparkles: remove snapshot metadata that can be inferred from path

### PR DESCRIPTION
**Pull request by Marigold**
Snapshot cleanup before the source -> origin refactoring.

* Determine fields `namespace`, `version`, `short_name` from filepath
* Remove confusing logic for inferring `version` from publication dates
* Remove it from walkthrough
* Test it out on a few snapshots

### CODE REVIEW

**Comment by Marigold** [comment]
@pabloarosado could you take a look when you have some time, please? It'd be nice to merge before larger metadata updates.

File: snapshots/dummy/2020-01-01/dummy.csv.dvc [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
   publication_date: '2020-01-01'
   source_name: Dummy short source citation
   source_published_by: Dummy full source citation
   url: https://www.url-dummy.com/
   source_data_url: https://raw.githubusercontent.com/owid/etl/master/walkthrough/dummy_data.csv
-  file_extension: csv
   license_url:
   license_name: ''
-  date_accessed: 2023-05-15
+  date_accessed: 2023-06-20
```
Comment (pabloarosado) [inline snapshots/dummy/2020-01-01/dummy.csv.dvc:10]: Was this changed by accident or did you re-run the snapshot?

**Review by pabloarosado — APPROVED** [review]
Nice work, this simplifies things. Please let me know if you foresee anything that could affect my improve-metadata-handling branch.

File: snapshots/dummy/2020-01-01/dummy.csv.dvc (hunk 1 above)
Comment (Marigold) [reply snapshots/dummy/2020-01-01/dummy.csv.dvc:10]: I'm occasionally using it for testing.

**Comment by Marigold** [comment]
> Please let me know if you foresee anything that could affect my improve-metadata-handling branch.

I hope it won't affect anything, but if you run into any issues, let me know.

**Pull request merged**